CHANGELOG.md

```diff
@@ -35,6 +39,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 https://tls.mbed.org/tech-updates/releases/mbedtls-2.16.5-and-2.7.14-released)
 release notes for the set of issues addressed.
 
+### Deprecation Notice
```

Comment:
I hesitate to make this official guidance lest people mob me for making the deprecation process too cumbersome. I chose a more conservative model for this particular feature to give people more time to switch over to the new way. I don't really want to take the guidance further than that.

Reply:
I agree on
> it might be clearer to say that we are deprecating the automatic inclusion of oecore EDL by default, and that they should use the COMPILE_SYSTEM_EDL flag to test this behavior ahead of time.

Reply:
I updated the CHANGELOG and pointed to the design doc for further guidance on what to do. let me know if that's what your were looking for.

Also please let me know if there's anything I can change in the doc for clarity.

Reply:
> I hesitate to make this official guidance lest people mob me for making the deprecation process too cumbersome. I chose a more conservative model for this particular feature to give people more time to switch over to the new way. I don't really want to take the guidance further than that.

To clarify, I'm not proposing that you make any kind of assertion about official guidance for the deprecation process in general, I'm asking how you envision the deprecation process of _this_ feature to proceed. The text asserts that the plan is:

- v0.9 does not break by default, supports `COMPILE_SYSTEM_EDL` for devs to experiment with importing EDLs manually.
- v0.10 breaks by default, removes `COMPILE_SYSTEM_EDL` entirely and devs must scramble to fix if they were not aware of the change in behavior previously.

I'm pointing out that is not an ideal developer experience. Many users do not read the CHANGELOG and will likely not be aware that there is any action needed if the builds transparently continue to work. This is why the existing two step deprecation plan breaks on the first edge:

- v0.9 breaks by default, but supports `COMPILE_SYSTEM_EDL` for devs to quickly get back to working builds while they deal with the long term switch over.
- v0.10 ends the grace period by removing the fallback `COMPILE_SYSTEM_EDL` mechanism entirely. 

I would suggest doing that if possible. If not, and it looks like internally we're not ready to make the default switch ourselves, I'm suggesting that this could benefit from the usual deprecation pattern:

- v0.9 does not break by default, supports `COMPILE_SYSTEM_EDL` for devs to experiment with importing EDLs manually.
- v0.10 breaks by default, but continues to support `COMPILE_SYSTEM_EDL` for devs to quickly get back to working builds while they deal with the long term switch over.
- v0.11 ends the grace period by removing the fallback `COMPILE_SYSTEM_EDL` mechanism entirely.

Reply:
Oh, that was not my intention. My intention for v0.10 was to do as you suggest and just change the default for `COMPILE_SYSTEM_EDL` from `ON` to `OFF`. I'll update the changelog to make that more clear.

Reply:
Updated CHANGELOG